### DISCUSSION TITLE
Static page with dynamic slug (From DatoCMS)

### DISCUSSION THREAD

**Discussion by vishalshah6153** [vercel/next.js#35522] [Help]
I have created a static Next JS application that gets data from DatoCMS using GraphQL. I have also enabled NextJs preview mode. Hence, I have created all routes inside pages folder, such as home, landing/[slug], net/[slug], and so forth.

I have some questions about this:
1). How can we deal with multilang when it's in preview mode and update the locale when it's opened through preview mode?
2) Is it possible to create a dynamic slug that contains no static text after domain
for example: domain/[slug] (dynamic slug). For now, I have created something like domain/static text(as as post)/[slug]
3) The current way I handle dynamic slugs and locales is to check coming slugs from CMS with existing pages and redirect based on the same. However, it has many issues when I modify the slug as well as the page name. Also, when changing the locale, the URL does not update.

In particular, I was worried about redirecting on frontend using the slugs added to the CMS. I am having difficulty dealing with dynamic slugs as well as multiple languages. 

Do you have any suggestions for how we can do this better?

**Comment by kelenakamura** [comment]
have you seen this starter? https://github.com/datocms/nextjs-demo-i18n - but i think this expects urls prefixed with /LANG/
